src/java/org/apache/cassandra/db/virtual/GossipStateTransitionsTable.java

```diff
@@ -0,0 +1,145 @@
+    private static final String INTERNAL_ADDRESS_AND_PORT = "internal_address_and_port";
+    private static final String NATIVE_ADDRESS_AND_PORT = "native_address_and_port";
+    private static final String STATUS_WITH_PORT = "status_with_port";
+    private static final String SSTABLE_VERSIONS = "sstable_versions";
+
+    /**
+     * Constructs a new {@link GossipStateTransitionsTable} with the given {@code keyspace}.
+     *
+     * @param keyspace the name of the keyspace
+     */
+    GossipStateTransitionsTable(String keyspace)
+    {
+        super(TableMetadata.builder(keyspace, TABLE_NAME)
+                           .comment(TABLE_COMMENT)
+                           .kind(TableMetadata.Kind.VIRTUAL)
+                           .partitioner(new LocalPartitioner(InetAddressType.instance))
+                           .addPartitionKeyColumn(ADDRESS, InetAddressType.instance)
+                           .addClusteringColumn(PORT, Int32Type.instance)
+                           .addClusteringColumn(TIMESTAMP, TimestampType.instance)
+                           .addRegularColumn(HOSTNAME, UTF8Type.instance)
+                           .addRegularColumn(GENERATION, Int32Type.instance)
+                           .addRegularColumn(HEARTBEAT, Int32Type.instance)
+                           .addRegularColumn(STATUS, UTF8Type.instance)
+                           .addRegularColumn(LOAD, UTF8Type.instance)
+                           .addRegularColumn(SCHEMA, UTF8Type.instance)
+                           .addRegularColumn(DC, UTF8Type.instance)
+                           .addRegularColumn(RACK, UTF8Type.instance)
+                           .addRegularColumn(RELEASE_VERSION, UTF8Type.instance)
+                           .addRegularColumn(REMOVAL_COORDINATOR, UTF8Type.instance)
+                           .addRegularColumn(INTERNAL_IP, UTF8Type.instance)
+                           .addRegularColumn(RPC_ADDRESS, UTF8Type.instance)
+                           .addRegularColumn(SEVERITY, UTF8Type.instance)
+                           .addRegularColumn(NET_VERSION, UTF8Type.instance)
+                           .addRegularColumn(HOST_ID, UTF8Type.instance)
+                           .addRegularColumn(TOKENS, UTF8Type.instance)
+                           .addRegularColumn(RPC_READY, UTF8Type.instance)
+                           .addRegularColumn(INTERNAL_ADDRESS_AND_PORT, UTF8Type.instance)
+                           .addRegularColumn(NATIVE_ADDRESS_AND_PORT, UTF8Type.instance)
+                           .addRegularColumn(STATUS_WITH_PORT, UTF8Type.instance)
+                           .addRegularColumn(SSTABLE_VERSIONS, UTF8Type.instance)
+                           .build());
+    }
+
+    /**
+     * @return a {@link org.apache.cassandra.db.virtual.AbstractVirtualTable.DataSet} with data populated with
+     * the {@code max_gossip_state_transitions_size} most recent gossip state transtitions
```

Comment:
This comment becomes obsolete when we change the tunable name 'max_gossip_state_transitions_size'. Probably good to not mention tunable name here?